src/PostgREST/App.hs

```diff
@@ -147,7 +147,7 @@ app dbstructure conf reqBody req =
       -- select * from public.proc(a := "foo"::undefined) where whereT limit limitT
 
     ([], _) -> do
-      body <- encode <$> tables (cs schema)
+      body <- encode <$> accessibleTables (filter ((== cs schema) . tableSchema) allTabs)
```

Comment:
What if `dbTables` was already filtered to be just the accessible ones? Or are all the tables needed in order to do the relation lookups?